include/SW_Domain.h

```diff
@@ -0,0 +1,26 @@
+#ifndef SWDOMAIN_H
+#define SWDOMAIN_H
+
+#include "include/SW_datastructs.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/* =================================================== */
+/*                  Local Definitions                  */
+/* --------------------------------------------------- */
+int* SW_DOM_calc_ncStartSuid(SW_DOMAIN* SW_Domain, int suid);
+void SW_DOM_calc_nSUIDs(SW_DOMAIN* SW_Domain);
+Bool SW_DOM_CheckProgress(char* domainType, int* ncStartSuid);
+void SW_DOM_CreateProgress(SW_DOMAIN* SW_Domain);
+void SW_DOM_read(char *InFiles[], SW_DOMAIN* SW_Domain, LOG_INFO* LogInfo);
+void SW_DOM_SetProgress(char* domainType, int* ncStartSuid);
+void SW_DOM_SimSet(SW_DOMAIN* SW_Domain, int userSUID, int nSUIDs,
```

Comment:
I believe we need to have a quick discussion on what to expect from the user suid. Because if this is an unsigned long (which is in my latest push), we cannot really detect a value less than zero unless we read in a long, check then convert it to unsigned. However, this will limit user input values compared to what the program uses.